libs/graph_utils/src/graph_utils.rs

```diff
@@ -114,15 +114,15 @@ pub fn generate_trace_graph_from_headers(
                 let mut values_iterator = statement.split("==");
                 let property_with_node = values_iterator.next().unwrap().clone().to_string();
                 let mut period_iterator = property_with_node.split(".");
-                let node = period_iterator.next().unwrap();
+                let node = period_iterator.next().unwrap().to_string();
                 let mut property = String::new();
                 for quality in period_iterator {
                     property.push_str(&quality);
                     property.push_str(".");
                 }
                 let _ = property.pop(); // get rid of trailing period
                 let node_weight: &mut (String, HashMap<String, String>) = graph
-                    .node_weight_mut(node_str_to_node_handle[node])
+                    .node_weight_mut(node_str_to_node_handle[&node])
```

Comment:
I do not think this is needed if you remove the to_string() call.

Reply:
That's true;  it's fixed now.